tests/system_tests/test_core/test_resume.py

```diff
@@ -99,3 +100,70 @@ def test_resume_output_log(wandb_backend_spy):
             if f.startswith("logs/output_") and f.endswith(".log")
         ]
         assert len(log_files) == 2
+
+
+def test_resume_config_preserves_image_mask(user, wandb_backend_spy):
+    img_array = np.zeros((2, 2, 3), dtype=np.uint8)
+    mask_array = np.zeros((1, 1), dtype=np.uint8)
+    mask_array[0, 0] = 1
+    class_labels = {1: "square"}
+
+    with wandb.init() as run:
+        run.log(
+            {
+                "test_image": wandb.Image(
+                    img_array,
+                    masks={
+                        "prediction": {
+                            "mask_data": mask_array,
+                            "class_labels": class_labels,
+                        }
+                    },
+                )
+            }
+        )
+
+    # Verify the config from the initial run
+    with wandb_backend_spy.freeze() as snapshot:
+        config = snapshot.config(run_id=run.id)
+        assert "_wandb" in config
+        wandb_config = config["_wandb"]["value"]
+
+        item_config = wandb_config.get("mask/class_labels", {})
+        initial_item_keys = set(item_config.keys())
+        assert len(initial_item_keys) > 0
```

Comment:
In tests, it's better to be explicit. Instead of organizing the test as Arrange-Act-Assert-Act-Assert, assert on the exact keys you expect in the final block (assuming that's possible). Do we expect `mask/class_labels` to contain `test_image` and `test_image_after_resume` keys? Why not just assert, at the end,

```
assert "test_image" in resumed_item_config
assert "test_image_after_resume" in resumed_item_config
```

The test would be much shorter and easier to read.